lib/improver/blending/weighted_blend.py

```diff
@@ -454,165 +454,216 @@ def process(self, cube, weights=None):
+    @staticmethod
+    def _check_all_time_points_equal(cube):
+        if cube.coords("time"):
+            time_points = cube.coord("time").points
+            if len(np.unique(time_points)) > 1:
+                msg = ("For blending using the forecast_reference_time "
+                       "coordinate, the points within the time coordinate "
+                       "need to be the same. The time points within the "
+                       "input cube are {}".format(time_points))
+                raise ValueError(msg)
+
+    def _check_weights_have_coord_shape(self, cube, weights):
+        if np.array(weights).shape != cube.coord(self.coord).points.shape:
+            msg = ('The weights array must match the shape '
+                   'of the coordinate in the input cube; '
+                   'weight shape is '
+                   '{0:s}'.format(np.array(weights).shape) +
+                   ', cube shape is '
+                   '{0:s}'.format(cube.coord(self.coord).points.shape))
+            raise ValueError(msg)
+
+    def _perform_blending(self, cube, weights):
+        cubelist = iris.cube.CubeList([])
+        for cube_thres in self._slices_over_threshold(cube):
+            cube_new = self._perform_blend_across_threshold(cube_thres,
+                                                            weights)
+            cube_new = conform_metadata(
+                cube_new, cube_thres, coord=self.coord,
+                cycletime=self.cycletime,
+                coords_for_bounds_removal=self.coords_for_bounds_removal)
+
+            cubelist.append(cube_new)
+        result = cubelist.merge_cube()
+        if isinstance(cubelist[0].data, np.ma.core.MaskedArray):
+            result.data = np.ma.array(result.data)
+        return result
+
+    def _perform_blend_across_threshold(self, cube_thres, weights):
+
+        # check to see if the data is percentile data
+        # if not, perc_coord will be None
+        perc_coord = self._get_perc_dim_coord(cube_thres)
+
+        # Blend the cube across the coordinate
+        if self.mode == "weighted_mean":
+            # Use percentile Aggregator if required
+            if perc_coord:
+                cube_new = self._blend_weighted_mean_percentile(cube_thres,
+                                                                weights,
+                                                                perc_coord)
+            # Else do a simple weighted average
+            else:
+                cube_new = self._blend_weighted_mean(cube_thres, weights)
+
+        # Else use the maximum probability aggregator.
+        elif self.mode == "weighted_maximum":
+            # If we have a percentile dimension and the mode is 'max' raise an
+            # exception.
+            if perc_coord:
+                msg = ('The "weighted_maximum" mode cannot be used with'
+                       ' percentile data.')
+                raise ValueError(msg)
+            else:
```

Comment:
`else` can be removed

Reply:
I agree. Do we have an overall IMPROVER style for this type of stuff?
Personally, I like to know which branch I've gone into, rather than having to track down any early exit etc.. but removing `else` would reduce the complexity, so I'm happy to remove this.

Reply:
I was really tempted to introduce something in the style guide, since the Google one doesn't cover it - but no, let's leave it as a personal opinion thing.